src/utils.py

```diff
@@ -1440,6 +1440,7 @@ def run_hill_climbing(
         check_goal: Callable[[_S], bool],
         get_successors: Callable[[_S], Iterator[Tuple[_A, _S, float]]],
         heuristic: Callable[[_S], float],
+        early_termination_heuristic_thresh: Optional[float] = 0,
```

Comment:
let's default this to None, rather than 0, but then change the PG3 approach to use 0 instead